### ISSUE TITLE
[Rx] Display button logic in Prescription card isn't working

### ISSUE THREAD

**Issue by webinista** [department-of-veterans-affairs/vets-website#2903]
Refill prescription button never gets displayed with current logic.

- When `is-refillable === true`: Display _Refill Prescription_ button
- When `is-trackable === true && is-refillable === false`: Display _Track package_ button
- When `is-trackable === false && is-refillable === false` : Display _Refill requested_ text
- When `is-refillable === false` and `refills === 0`: Display _Message my provider_ link.

### Questions
- Does `is-refillable` ever become `false` while `is-trackable` is true? For example: is there a window of time after a refill is requested when it can't be requested again?

Blocks #2576, #2630

**Comment by ayaleloehr** [comment]
Yes to your question. Directly after refilling a prescription, is-refillable will become false, but is-trackable will stay true (if it was already true).   So I would say the _track package_ button (bullet 2 above) should be shown when `is-trackable === true` and `is-refillable === false`. 

And now that I'm thinking of it, we don't actually have the tracking information/URL on the active prescriptions page, so the _track package_ button should probably take you to the prescription detail page, where you can actually see the packages that have been shipped (would be ideal if it can auto scroll you down the page to that section maybe?)

**Comment by ayaleloehr** [comment]
Also, I don't believe bullet 3 is correct.  @gnakm do you already have these rules written out somewhere?  We're definitely going to need to be certain of this logic.

**Comment by gnakm** [comment]
@ayaleloehr You're right 3rd bullet is incorrect. I might need your help explaining this logic. 

When a user clicks the refillable button and then confirms to prescription, the refill should no longer be refillable but also tracking information will not be readily available, therefore, there should be a Refill requested text.

When `is-trackable === false` and `is-refillable === false` : Display Refill requested text

I might totally be getting this wrong but `is-refillable` accounting for ALL the refills that are left? Alex, could you verify which situation makes more sense: 
 
1. You submitted refilled your 2nd refill (out of 5) but there are still 3 refills left therefore the Rx is still refillable `is-refillable === true' 

_or_

2. You submitted refilled your 2nd refill (out of 5) but because you just submitted a refill, the Rx is currently not refillable. `is-refillable === false' 

Also I had the logics down somewhere. Need to step out. Will add when I am back!

**Comment by webinista** [comment]
@ayaleloehr The [API docs](https://github.com/department-of-veterans-affairs/prescriptions-api/blob/master/api-docs/prescriptions-active.md) say that there's a tracking link included with each active prescription. Are you sure that's not the case?

**Comment by ayaleloehr** [comment]
@webinista you're correct that there is a tracking link there, but I think it means something different than what you're expecting; that is a link to an API request to our vets.gov API to return  all the tracking information for that prescription (doc [here](https://github.com/department-of-veterans-affairs/prescriptions-api/blob/master/api-docs/prescriptions-prescription_id-tracking.md) on what's returned).  I believe what you were expecting was a link to UPS/USPS directly (correct me if that's wrong!). 

So what I was trying to say was that from the Active page, we can't send you directly to a USPS/UPS link, because those aren't available there (and we would have to make a separate API call to get it for each prescription, which we don't want to do). Instead when you click that "track package" button (might need a new name now?) it should take you to the prescription detail page, which is the place that the actual links to UPS/USPS exist. 

Is that helpful?  Open to thoughts!

**Comment by ayaleloehr** [comment]
@gnakm I think it's the second one you wrote; "because you just submitted a refill, the Rx is currently not refillable."  Can you get this logic all organized into one comment when you have a chance so it's in a single place?  Happy to talk through it with you if that would be helpful.

**Comment by gnakm** [comment]
@ayaleloehr @webinista 

When `is-refillable === true`: Display Refill Prescription button

When `is-refillable === false && is-trackable === true`: Display 
Track package button

When `is-refillable === false && is-trackable === false` : Display Refill requested text

When `is-refillable === false and refills === 0`: Request renewal.

When `is-refillable === false && refillStatus === Suspended`: Display Call Provider.

When `is-refillable === false && refillStatus === On hold`: Display Call Provider.

Also can be found [here](https://github.com/department-of-veterans-affairs/kudos-team/blob/master/Prescriptions/design/Rx%20statuses.md).

**Comment by ayaleloehr** [comment]
@gnakm sorry I didn't think of this before...if `is-refillable === false` && `is-trackable === true` && `refills === 0`, do we display both "Track package" and "Request renewal" or just one of them (and if only one, which one?)

**Comment by webinista** [comment]
It sounds like we really have two major statuses: Refillable / not refillable. And if it's not refillable, there are several "sub-statuses" it could have. I'll come back to this point.

**Comment by U-DON** [comment]
Regarding 'track package' button behavior: What if upon clicking the button, we make a request for the tracking info of that prescription, grab the tracking URL from the first (most recent) tracking info, and load that in a new tab?

Since we don't have that URL available with `GET /prescriptions/active`, this would require an extra request, but it be a direct step for the user to the package tracking info.

**Comment by webinista** [comment]
So I think the logic should look more like this (using pseduo-code):

```
if (is-refillable === true):
   Show _Refill prescription_ button
else:

  if (is-trackable === true):
    Show _Track package_ button
  else:
   Show _Refill requested_ text


  if (refills-remaining === 0):
    Show _Request renewal_ text


  if(refill-status === 'on hold' or 'suspended'):
    Show _Call provider_ text

```

### Questions:

- Is _Request renewal_ a link to secure messaging? Or is this the same as _Call provider_?
- We have six possible refill statuses according to what's in the glossary:
   - Active
   - Discontinued
   - Expired
   - On hold
   - Pending
   - Suspended
  
  We accounted for _On hold_ and _Suspended_ above. But: 
  - What should when the status is _Expired_ or _Discontinued_?
  - What should happen when the status is _Pending_?
  
  Maybe the last block looks like this instead?
  
  ```
  if(refill-status !== 'active' && refill-status !== 'pending')
  	Show _Call provider_ text
  endif
  ```

**Comment by gnakm** [comment]
- I just received terminology docs from a stakeholder. 
It's the 6  (Active, Discontinued, Expired, On hold, Pending, Suspended) + Unknown. So 7 in total. 
- Eventually Call provider & Request Renewal will become Secure Messaging Links.   
- Expired, Discontinued and Unknown are no longer active and they all should say "Call provider." but be in the Order History page. 
- Pending should say Refill Requested

**Comment by webinista** [comment]
(Probably) resolved in 8d5dc5d
